azure-pipelines.yml

```diff
@@ -17,7 +17,7 @@ variables:
   CIBW_BUILD_VERBOSITY: "2"
   CIBW_BEFORE_BUILD: "pip install -U numpy Cython jupyter ipywidgets"
   CIBW_BEFORE_BUILD_MACOS: "npm install npm@latest -g; pip install -U pip setuptools"
-  CIBW_BEFORE_BUILD_LINUX: "yum install -y fontconfig xvfb; pip install -U pip setuptools; if [ `getconf LONG_BIT` == '64' ]; then pip install https://files.pythonhosted.org/packages/f2/00/6f332e63b33d24dc3761916e6d51402a7a82dd43c6ca8a96e24dda32c6b5/freetype_py-2.1.0.post1-py2.py3-none-manylinux1_x86_64.whl; else pip install https://files.pythonhosted.org/packages/b8/ea/adfbeb3762a03c1d1c1ff751101efb1fddb7aadffa991d39bdb486d0557e/freetype_py-2.1.0.post1-py2.py3-none-manylinux1_i686.whl; fi"
+  CIBW_BEFORE_BUILD_LINUX: "yum install -y fontconfig; pip install -U pip setuptools; if [ `getconf LONG_BIT` == '64' ]; then pip install https://files.pythonhosted.org/packages/f2/00/6f332e63b33d24dc3761916e6d51402a7a82dd43c6ca8a96e24dda32c6b5/freetype_py-2.1.0.post1-py2.py3-none-manylinux1_x86_64.whl; else pip install https://files.pythonhosted.org/packages/b8/ea/adfbeb3762a03c1d1c1ff751101efb1fddb7aadffa991d39bdb486d0557e/freetype_py-2.1.0.post1-py2.py3-none-manylinux1_i686.whl; fi"
```

Comment:
Is `xvfb` not available? We may want to have this in the future if we run actual tests on Azure.

Reply:
it fail on centos7 docker image. I do not see anny difference in number of executed test. So I remove this instead of workaround. I can work on this.

I have other question. Why you install something with pip in this steep? This step is executed onece on begin and does not affect all python interpreters.

Reply:
If I remember correctly it was that some of the vispy cython extensions need freetype and the freetype_py package (which we also use) had it bundled with. According to the docs this is run before each wheel build: https://cibuildwheel.readthedocs.io/en/stable/options/#before-build

Reply:
Oh and we install this specific version of freetype_py because for some reason `pip` wasn't giving us the right build when run as `pip install freetype-py`.

Reply:
Ok. My bad. I wrong remember execution order in cibuildwheel. When https://github.com/joerick/cibuildwheel/issues/54 will be finished then yum call can be executed onece.

I test with `pip install freetype-py` and build pass https://dev.azure.com/bokota/open%20source%20contrib/_build/results?buildId=419&view=results

Second observation. manylinux2014 produce wheel which is compatibile with manylinux1. So maybe not add manylinux2014 build but switch to it?

Reply:
Unless I'm wrong, manylinux2014 is for CentOS7 and above. Switching to only using 2014 would mean the wheels would no longer work/be available for CentOS 6.

Reply:
If you want to replace the freetype-py complicated if statement with the simpler pip install that would be fine with me.

Reply:
Do you know of a popular python package that is building linux wheels on something other than manylinux1 right now?

Reply:
if you look at result of manylinux2014 https://dev.azure.com/vispy/vispy/_build/results?buildId=369&view=artifacts&type=publishedArtifacts then you see that it produce also manylinux1. It works in this way because auditwheel checks if wheel is manylinux1 compatible. And if it is then produce two wheels. Also manylinux2014 is faster.

Reply:
Ok so you are saying if the `manylinux1` specific job is removed and `manylinux2014` is the *only* linux environment, then it should produce (and upload) a `manylinux2014` wheel *and* a `manylinux1` wheel? But that *only* happens if it is compatible right?

Reply:
in current configuration you not have manylinux1 job. You have manylinux2010 job. 

currently manylinux1 wheel is produced only if manylinux2010 wheel is compatibile with manylinux1. 

If you would like to have manylinux1 job you need to use:
```
  variables:
    CIBW_MANYLINUX_X86_64_IMAGE: manylinux1
    CIBW_MANYLINUX_I686_IMAGE: manylinux1
```
in build specification. 

In this PR configuration. Both manylinux2010 and manylinux2014 produces manylinux1 wheel.

Reply:
Ah ok. So the question is then do we turn off the 2010 job and only have the 2014 job, right? Since the 2014 job is producing manylinux1 wheels that must mean they are compatible? Which means they'll work on CentOS 5+. That's ok with me then.

Reply:
current PR configuration produce `manylinux1`, `manylinux2010` and `manylinux2014` maybe `maylinux2010` is not needed. 

To be sure that `manylinux1`is produced something like this may be added.
```bash
ls *cp36*manylinux1* &&  ls *cp37*manylinux1* &&  ls *cp38*manylinux1* 
```

Reply:
Since it seems like `manylinux1` is currently being produced and our cython code rarely gets updated, I'm not too worried about making sure it is always created. This is with the assumption that if it ever stops getting produced (because it isn't compatible anymore) that CentOS5 will be long dead and CentOS6 will also be deprecated. I think you can get rid of the manylinux2010 based on what we've talked about here.

Reply:
ready

Reply:
Yep. Thanks again.